modin/pandas/dataframe.py

```diff
@@ -2523,28 +2522,9 @@ def __setitem__(self, key, value):
             if isinstance(value, Series) and len(self.columns) == 0:
                 self._query_compiler = value._query_compiler.copy()
                 # Now that the data is appended, we need to update the column name for
-                # that column to `key`, otherwise the name could be incorrect. Drop the
-                # last column name from the list (the appended value's name and append
-                # the new name.
-                self.columns = self.columns[:-1].append(pandas.Index([key]))
```

Comment:
This `columns` variable is empty, the code will look simpler if we do not call an unnecessary `append` op.

Reply:
I don't think thats the case here @dchigarev because then the `len(self.columns) == 0` would be false, right?

Reply:
I found a producer for the behavior @dchigarev was talking about. Fixed and added test.